hathor/builder/cli_builder.py

```diff
@@ -293,11 +300,15 @@ def create_manager(self, reactor: Reactor) -> HathorManager:
             whitelist_only=False,
             rng=Random(),
         )
-        p2p_manager.add_sync_factory(SyncVersion.V1_1, SyncV11Factory(p2p_manager))
-        p2p_manager.add_sync_factory(SyncVersion.V2, SyncV2Factory(p2p_manager))
-        if enable_sync_v1:
+        # sync-v1 support:
+        if sync_v1_support > SyncSupportLevel.UNAVAILABLE:
+            p2p_manager.add_sync_factory(SyncVersion.V1_1, SyncV11Factory(p2p_manager))
+        if sync_v1_support is SyncSupportLevel.ENABLED:
             p2p_manager.enable_sync_version(SyncVersion.V1_1)
-        if enable_sync_v2:
+        # sync-v2 support:
+        if sync_v2_support > SyncSupportLevel.UNAVAILABLE:
+            p2p_manager.add_sync_factory(SyncVersion.V2, SyncV2Factory(p2p_manager))
+        if sync_v2_support is SyncSupportLevel.ENABLED:
             p2p_manager.enable_sync_version(SyncVersion.V2)
```

Comment:
This code is duplicate from the builder, should we create a method somewhere?

Reply:
Moved to a classmethod `SyncSupportLevel.add_factories`.

Reply:
Thanks!